exercises/src/test/scala/fpinscala/iomonad/IOSpec.scala

```diff
@@ -0,0 +1,42 @@
+package fpinscala.iomonad
+
+import org.specs2.mutable._
+import org.specs2.matcher.Scope
+
+class IOSpec extends Specification {
+
+  class IO3Fixture extends Scope {
+    import IO3._
+    import Console._
+
+    def countDown(from: Int): Free[Console, Unit] = {
+      if (from > 0)
+        for {
+          _ <- printLn(s"$from...")
+          _ <- countDown(from - 1)
+        } yield ()
+      else
+        printLn("0.")
+    }
+  }
+
+  "runConsoleFunction0" >> {
+    "is not stack-safe" >> new IO3Fixture {
+      import IO3._
+
+      runConsoleFunction0(countDown(100000)).apply should throwAn[StackOverflowError]
```

Comment:
two things: 
1) ackerman function usage example.
2) if you wanted to create a matcher like that you need to invert the order to make it work.